drivers/flash/nrf_qspi_nor.c

```diff
@@ -534,19 +534,47 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 			  size_t size)
 {
+	const void *sptr = src;
+	size_t dlen = size;
+	u8_t buf[4];
+
 	if (!src) {
 		return -EINVAL;
 	}
 
-	/* write size must be non-zero multiple of 4 bytes */
-	if (((size % 4U) != 0) || (size == 0)) {
-		return -EINVAL;
-	}
 	/* address must be 4-byte aligned */
 	if ((addr % 4U) != 0) {
 		return -EINVAL;
 	}
 
+	/* read size must be non-zero multiple of 4 bytes */
+	if (size < 4U) {
+		src = buf;
+		size = sizeof(buf);
+		nrfx_err_t res = nrfx_qspi_read(buf, size, addr);
+
+		if (res != 0) {
+			return qspi_get_zephyr_ret_code(res);
+		}
+		memcpy(buf, sptr, dlen);
+	} else if (((size % 4U) != 0) || (size == 0)) {
+		return -EINVAL;
+	}
+
+#if IS_ENABLED(NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH)
```

Comment:
Do we need to have this configurable? Is there situations when you don't want this feature?

Also, is it possible to avoid CPP here? `if (IS_ENABLED) ...` etc.

Reply:
Needed to ensure that we don't change the driver to suddenly use tons of stack instead of faulting.